pkg/manager/impl/execution_manager.go

```diff
@@ -624,9 +626,7 @@ func resolvePermissions(ctx context.Context, request *admin.ExecutionCreateReque
 		logger.Warning(ctx, "AssumableIamRole not set.")
 	}
 
-	if request.Spec.AuthRole != nil && len(request.Spec.AuthRole.KubernetesServiceAccount) > 0 {
```

Comment:
this is overridable in the request so we prefer to read the value there if it's set

Reply:
This is the workflow that I am testing with:
```
from flytekit.common.launch_plan import SdkLaunchPlan
from flytekit.sdk.tasks import dynamic_task
from flytekit.sdk.workflow import workflow_class


@dynamic_task
def launch_simple_workflow(wf_params):
    launch_plan = SdkLaunchPlan.fetch(
        project='flytefunctionaltests',
        domain='production',
        name='tests.common.workflows.simple.SimpleWorkflow',
    )

    launch_plan.assign_name('ctsai')

    execution = launch_plan(input_1=1)
    yield execution


@workflow_class
class DynamicTaskWorkflow(object):
    task = launch_simple_workflow()
```

Reply:
It seems like the parent task sets the auth role in the request so it overrides the auth role defined in the launch plan

Reply:
nvm